File: cpp/OperatingSystem.cpp
#pragma once
#include <iostream>
#include <vector>
using namespace std;

#include "Software.cpp"

class OperatingSystem : public Software {
    private:
        string architecture;
        string kernelType;
    
    public:
        OperatingSystem(string n, string v, string arch, string kernel)
            : Software(n, v), architecture(arch), kernelType(kernel) {}
    
        // Getters
        string getArchitecture() { return architecture; }
        string getKernelType() { return kernelType; }
    
        // Setters
        void setArchitecture(string arch) { architecture = arch; }
        void setKernelType(string kernel) { kernelType = kernel; }
    };

File: java/program/OperatingSystem.java
public class OperatingSystem extends Software {
    private String architecture;
    private String kernelType;

    public OperatingSystem(String name, String version, String architecture, String kernelType) {
        super(name, version);
        this.architecture = architecture;
        this.kernelType = kernelType;
    }

    // Getters
    public String getArchitecture() {
        return architecture;
    }

    public String getKernelType() {
        return kernelType;
    }

    // Setters
    public void setArchitecture(String architecture) {
        this.architecture = architecture;
    }

    public void setKernelType(String kernelType) {
        this.kernelType = kernelType;
    }
}
